File: bots/pythonClient/Bots/aiFactory.py
from Bots.bot_ai import BotAI
from Bots.first_ai import FirstAI
from Bots.bodennuckel import BodenNuckel

ai_bots = {
    "FirstAI": FirstAI,
    "BodenNuckel": BodenNuckel,
}


def ai_factory(bot_selection="MyAI") -> BotAI:
    return ai_bots[bot_selection]()

File: bots/typescriptClient/src/bots/aiFactory.ts
import { BotAI } from './botAI';
import { FirstAI } from './firstAI';
import { MyAI } from './myAI';

const aiBots: Record<string, new () => BotAI> = {
  'FirstAI': FirstAI,
  'MyAI': MyAI,
};

export function aiFactory(botSelection: string = 'MyAI'): BotAI {
  const BotClass = aiBots[botSelection];
  if (!BotClass) {
    throw new Error(`Bot type '${botSelection}' not found`);
  }
  return new BotClass();
} 